Glims SampleSheet Generator

Istaisa opened the pull request:
Initial code for the GLIMS SampleSheet generator. Due to everything being stored in LIMS, it now just reformats the TSV input to a samplesheet output.

In this pull request:
- New module for GLIMS samplesheet generator - renamed existing module in homepage to make it clear which is which
- Parses what is, as far as I know, how GLIMS queries are currently formatted (TSV with set fields, TSO500 and WES example included in repository)
- No models for samplesheet data as this is stored in LIMS, but fixture added for assay (needs installing before testing)
- Should correctly format samplesheets including family structure, and WES being on the NextSeq or NovaSeq
- Unit tests for the GlimsSample class

Needs testing:
- Do these samplesheets match the format for existing samplesheets
- Will data demultiplex from these samplesheets

Still to do:
- Model and method to parse referrals - right now I don't know how the codes map
- Return a list of warnings where the samplesheet is not correct but can still be loaded (e.g. referral missing, family structure incorrect)
- Make it pretty
- Update README with instructions for fixtures and admin management information for tests
- Make schema
- Redirect page after samplesheet download

millardkieron commented:
Code makes sense. New module clear for GLIMS. Test tsv files upload and reformat okay (TSO on NovaSeq and WES on NovaSed and NextSeq). All unit tests ran okay.

millardkieron reviewed (approved):
All okay.

Istaisa commented:
- Added code and tests to parse samplesheet referrals - still waiting on a list of codes and panels but we can add this closer to the time, I've added some mock data in the interim
- It looks slightly nicer now
- Error messages are more informative

millardkieron reviewed (approved):
Load up and tests worked all okay. Loaddata lims assay and referral type worked all okay. Expected error message in yellow bar shown when error samplesheet uploaded. Code layout and logic all okay.

Istaisa commented:
Closing this as new samplesheet generator is now independent app
